keras_cv/models/backbones/mobilenet_v3/mobilenet_v3_backbone.py

```diff
@@ -0,0 +1,442 @@
+                of filters in each layer.
+            - If `alpha` = 1, default number of filters from the paper
+                are used at each layer.
+        minimalistic: in addition to large and small models, this module also
+            contains so-called minimalistic models; these models have the same
+            per-layer dimensions characteristic as MobilenetV3 however, they
+            don't utilize any of the advanced blocks (squeeze-and-excite units,
+            hard-swish, and 5x5 convolutions). While these models are less
+            efficient on CPU, they are much more performant on GPU/DSP.
+        dropout_rate: a float between 0 and 1 denoting the fraction of input
+            units to drop, defaults to 0.2.
+
+    Examples:
+    ```python
+    input_data = tf.ones(shape=(8, 224, 224, 3))
+
+    # Randomly initialized backbone with a custom config
+    model = MobileNetV3Backbone(
+        stack_fn=None,
+        last_point_ch=1024,
+        include_rescaling=False,
+    )
+    output = model(input_data)
+    ```
+    """
+
+    def __init__(
+        self,
+        *,
+        stack_fn,
+        last_point_ch,
+        include_rescaling,
+        input_shape=(None, None, 3),
+        input_tensor=None,
+        alpha=1.0,
+        minimalistic=True,
+        dropout_rate=0.2,
+        **kwargs,
+    ):
+        if minimalistic:
+            kernel = 3
+            activation = layers.ReLU()
+            se_ratio = None
+        else:
+            kernel = 5
+            activation = apply_hard_swish
+            se_ratio = 0.25
+
+        inputs = utils.parse_model_inputs(input_shape, input_tensor)
+        x = inputs
+
+        if include_rescaling:
+            x = layers.Rescaling(scale=1 / 255)(x)
+
+        x = layers.Conv2D(
+            16,
+            kernel_size=3,
+            strides=(2, 2),
+            padding="same",
+            use_bias=False,
+            name="Conv",
+        )(x)
+        x = layers.BatchNormalization(
+            axis=channel_axis,
+            epsilon=1e-3,
+            momentum=0.999,
+            name="Conv/BatchNorm",
+        )(x)
+        x = activation(x)
+
+        x = stack_fn(x, kernel, activation, se_ratio)
+
+        last_conv_ch = depth(backend.int_shape(x)[channel_axis] * 6)
+
+        # if the width multiplier is greater than 1 we
+        # increase the number of output channels
+        if alpha > 1.0:
+            last_point_ch = depth(last_point_ch * alpha)
+        x = layers.Conv2D(
+            last_conv_ch,
+            kernel_size=1,
+            padding="same",
+            use_bias=False,
+            name="Conv_1",
+        )(x)
+        x = layers.BatchNormalization(
+            axis=channel_axis,
+            epsilon=1e-3,
+            momentum=0.999,
+            name="Conv_1/BatchNorm",
+        )(x)
+        x = activation(x)
+
+        super().__init__(inputs=inputs, outputs=x, **kwargs)
+
+        self.stack_fn = stack_fn
```

Comment:
In order to use this as a `Backbone` for Object detection, we'll also need to setup `self.pyramid_level_outputs`.

This will most likely require updating the `stack_fn` to return `x` alongside the pyramid_level_outputs.

The `pyramid_level_outputs` should be a dict with keys 1...N where the values are the last intermediate outputs of the model where the size of the Tensor has been reduced by 2^N.

(So, for example, if the input is 512x512, the key 1 would point to the output of the last intermediate res block whose output is 256x256, and the key 3 would point to the output of the last intermediate res block whose output is 64x64)

Reply:
Is it obvious how to do this? I think the `stack_fn` will need to be turned into a for loop or otherwise produce this for the backbone model. If we can make this look more like the resnet implementations it would be ideal, but otherwise let's find a workable delegation of responsibility here.

Reply:
Done